src/core/qgsvectorlayer.cpp

```diff
@@ -1957,12 +1978,32 @@ bool QgsVectorLayer::writeSymbology( QDomNode& node, QDomDocument& doc, QString&
 {
   ( void )writeStyle( node, doc, errorMessage );
 
-  // FIXME
-  // edittypes are written to the layerNode
-  // by slot QgsEditorWidgetRegistry::writeMapLayer()
-  // triggered by signal QgsProject::writeMapLayer()
-  // still other editing settings are written here,
-  // although they are not part of symbology either
+  QDomElement fieldConfigurationElement = doc.createElement( QStringLiteral( "fieldConfiguration" ) );
+  node.appendChild( fieldConfigurationElement );
+
+  int index = 0;
+  Q_FOREACH ( const QgsField& field, mFields )
+  {
+
+    QDomElement fieldElement = doc.createElement( QStringLiteral( "field" ) );
+    fieldElement.setAttribute( QStringLiteral( "name" ), field.name() );
+
+    fieldConfigurationElement.appendChild( fieldElement );
+
+    QgsEditorWidgetSetup widgetSetup = field.editorWidgetSetup();
+
+    // TODO : wrap this part in an if to only save if it was user-modified
```

Comment:
Nope, but implementation of this would require changes to the configuration dialog to explicitly specify where something is overridden and where it's left to default. That's out of scope for this project.